# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/nsBaseDragService.cpp
# Commit: beeda5ef3712
# Full Hash: beeda5ef3712b73a19c7a4ce08a52ab4dd08b802
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 20 - Move EndDragSession from nsIDragService to nsIDragSession r=gstoll,rkraesig,win-reviewers,geckoview-reviewers,m_kato
#   
#   This is a straightforward move.  It does take the liberty of breaking out an
#   EndDragSessionImpl method, which will be needed later.
#   
# ==============================================================================

diff -r 6c3ffc8c534c -r beeda5ef3712 widget/nsBaseDragService.cpp
--- a/widget/nsBaseDragService.cpp	Thu Jul 04 07:48:10 2024 +0000
+++ b/widget/nsBaseDragService.cpp	Thu Jul 04 07:48:11 2024 +0000
@@ -73,9 +73,7 @@
 }
 
 nsBaseDragService::nsBaseDragService()
-    : mEndingSession(false),
-      mHasImage(false),
-      mContentPolicyType(nsIContentPolicy::TYPE_OTHER),
+    : mContentPolicyType(nsIContentPolicy::TYPE_OTHER),
       mSuppressLevel(0) {}
 
 nsBaseDragService::~nsBaseDragService() = default;
@@ -397,7 +395,10 @@
     // Set mDoingDrag so that EndDragSession cleans up and sends the dragend
     // event after the aborted drag.
     mDoingDrag = true;
-    EndDragSession(true, 0);
+    RefPtr<nsIDragSession> session = GetCurrentSession(aWidget);
+    if (session) {
+      session->EndDragSession(true, 0);
+    }
   }
 
   return rv;
@@ -612,14 +613,19 @@
 
 //-------------------------------------------------------------------------
 NS_IMETHODIMP
-nsBaseDragService::EndDragSession(bool aDoneDrag, uint32_t aKeyModifiers) {
+nsBaseDragSession::EndDragSession(bool aDoneDrag, uint32_t aKeyModifiers) {
+  return EndDragSessionImpl(aDoneDrag, aKeyModifiers);
+}
+
+nsresult nsBaseDragSession::EndDragSessionImpl(bool aDoneDrag,
+                                               uint32_t aKeyModifiers) {
   if (!mDoingDrag || mEndingSession) {
     return NS_ERROR_FAILURE;
   }
 
   mEndingSession = true;
 
-  if (aDoneDrag && !mSuppressLevel) {
+  if (aDoneDrag && !GetSuppressLevel()) {
     FireDragEventAtSource(eDragEnd, aKeyModifiers);
   }
 